common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -107,6 +109,11 @@ func NewVisibilityStore(
 	}
 }
 
+func newReadContext() context.Context {
+	ctx, _ := context.WithTimeout(context.Background(), readTimeout)
+	return ctx
+}
+
```

Comment:
probably need to used the cancel function? e.g. timeout is pretty much like a timer